concord_cli/kill_task.py

```diff
@@ -1,49 +1,184 @@
+    ip = get_zookeeper_master_ip(zookeeper, zk_path)
     logger.info("Found leader at: %s" % ip)
     (addr, port) = ip.split(":")
     logger.info("Initiating connection to scheduler")
     cli = get_sched_service_client(addr,int(port))
     logger.info("Sending request to scheduler")
-    try: cli.killTask(options.task_id)
+    try:
+        for task_id in task_ids:
+            logger.info('Sending request to kill task: %s', task_id)
+            cli.killTask(task_id)
     except thrift.Thrift.TApplicationException as e:
-        logger.error("Error killing task:%s" % options.task_id)
+        logger.error("Error killing task: %s" % task_id)
         logger.exception(e)
     logger.info("Done sending request to server")
 
+def collect_taskids(zookeeper):
+    """ Querys zk for topology metadata and returns a list of task_ids"""
+    meta = get_zookeeper_metadata(zookeeper)
+    if meta is None or len(meta.computations) == 0:
+        return []
+
+    # Obtain taskids from list of nodes from list of computations then flatten
+    return flatten([list(d.taskId for d in c.nodes) \
+                    for c in meta.computations.values()])
+
+def parse_input(iterable, user_input):
+    """ Parses user input and either will abort or return the data at
+    a given index or indicies or range"""
+    def token_expand(token):
+        if token.isdigit():
+            return [int(token)]
+        # Otherwise, we are dealing with a range
+        inclusive_range = token.split('..')
+        if len(inclusive_range) == 2:
+            return range(int(inclusive_range[0]), int(inclusive_range[1]) + 1)
+        raise Exception('Unexpected token detected')
+
+    tokens = user_input.split(',')
+    if len(tokens) == 1:
+        if tokens[0] == 'quit' or tokens[0] == 'q':
+            sys.exit(1)
+        elif tokens[0] == 'all' or tokens[0] == 'a':
+            return iterable
+
+    # Expand ranges, flatten results, then insert into set to remove dups
+    indicies = set(flatten(map(token_expand, tokens)))
+
+    # Returns desired choices performing bounds check at the same time
+    def verify_index(index):
+        if index < 1 or index > len(iterable):
+            raise Exception('Index out of bounds')
+        return iterable[index-1]
+    return map(verify_index, indicies)
+
+def prompt_selection(iterable, heading, printer):
+    """ Takes 'iterable' and displays it in an ascii table using printable
+    method to determine how to transform the data. Will also prompt user for
+    a selection and return the blob to the caller"""
+    def preprinter(value_pair):
+        idx, data = value_pair
+        return [str(idx)] + printer(data)
+
+    # Create table
+    heading.insert(0, 'Idx')
+    table_data = zip(xrange(1, len(iterable) + 1), iterable)
+    table_data = map(preprinter, table_data)
+    table_data.insert(0, heading)
+
+    # Print table in nice format, wait for user selection
+    print AsciiTable(table_data).table
```

Comment:
this is fly!
You should probably flush the screen before. 

maybe ` for i in range(0,100): print "" `